### PR TITLE
Tables.jl integration

### PR DESCRIPTION
**Pull request by iblis17**
- [x] doc
- [x] NEWS

### CODE REVIEW

**Comment by femtotrader** [comment]
You should probably add some unit tests

**Comment by iblis17** [comment]
will do, but before that, there are still some issue with the Tables that need to be discuss. I will file it on Tables.jl later.

**Comment by femtotrader** [comment]
Please don't forget to link to this issue from https://github.com/JuliaData/Tables.jl to keep a track.
Thanks

File: src/tables.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Table.Schema(colnames(ta), ntuple(x -> T, ndims(ta)))
+
+
+function TimeArray(x; on)
+    Tables.istable(x) || throw(ArgumentError("TimeArray requires a table input"))
+
+    sch = Tables.schema(x)
+    names = sch.names
+    (on ∉ names) && throw(ArgumentError("time index `$on` not found"))
+    names′ = [x for x ∈ sch.names if x != on]
```
Comment (ararslan) [inline src/tables.jl]: This can be `filter(!isequal(on), sch.names)`

**Comment by iblis17** [comment]
ref: https://github.com/JuliaData/Tables.jl/issues/40

**Comment by iblis17** [comment]
Ready for review.

**Pull request merged**